libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,507 @@
+        }
+    }
+
+    if( TransportSocketStatus == SOCKETS_ERROR_NONE )
+    {
+        /* Secure Sockets uses TickType_t therefore replace timeout vale with portMAX_DELAY if it exceeds. */
+        if( pdMS_TO_TICKS( sendTimeoutMs ) > portMAX_DELAY )
+        {
+            sendTimeout = portMAX_DELAY;
+        }
+        else
+        {
+            sendTimeout = pdMS_TO_TICKS( sendTimeoutMs );
+        }
+
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_SNDTIMEO,
+                                                    &sendTimeout,
+                                                    sizeof( TickType_t ) );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to set socket send timeout. Socket status %d.", TransportSocketStatus ) );
+        }
+    }
+
+    return TransportSocketStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static TransportSocketStatus_t establishConnect( NetworkContext_t * pNetworkContext,
+                                                 const ServerInfo_t * pServerInfo,
+                                                 const SocketsConfig_t * pSocketsConfig )
+{
+    Socket_t tcpSocket = ( Socket_t ) SOCKETS_INVALID_SOCKET;
+    TransportSocketStatus_t returnStatus = TRANSPORT_SOCKET_STATUS_SUCCESS;
+    int32_t TransportSocketStatus = ( int32_t ) SOCKETS_ERROR_NONE;
+    SocketsSockaddr_t serverAddress = { 0 };
+    const size_t hostnameLength = pServerInfo->hostNameLength;
+
+    if( ( hostnameLength > ( size_t ) MAX_DNS_NAME_LENGTH ) )
+    {
+        LogError( ( "Host name length %d exceeds max length %d",
+                    hostnameLength, MAX_DNS_NAME_LENGTH ) );
+        returnStatus = TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER;
+    }
+
+    if( returnStatus == TRANSPORT_SOCKET_STATUS_SUCCESS )
+    {
+        /* Create a new TCP socket. */
+        tcpSocket = SOCKETS_Socket( SOCKETS_AF_INET,
+                                    SOCKETS_SOCK_STREAM,
+                                    SOCKETS_IPPROTO_TCP );
+
+        if( tcpSocket == ( Socket_t ) SOCKETS_INVALID_SOCKET )
+        {
+            LogError( ( "Failed to create new socket.\n" ) );
+            returnStatus = TRANSPORT_SOCKET_STATUS_INSUFFICIENT_MEMORY;
+        }
+    }
+
+    if( returnStatus == TRANSPORT_SOCKET_STATUS_SUCCESS )
+    {
+        /* Set up connection encryption if credentials are provided. */
+        if( pSocketsConfig->enableTls )
+        {
+            if( ( int32_t ) SOCKETS_ERROR_NONE != tlsSetup( pSocketsConfig,
+                                                            tcpSocket,
+                                                            ( uint8_t * ) pServerInfo->pHostName,
+                                                            hostnameLength ) )
```

Comment:
How about updating `tlsSetup` to take in a `const char *` instead and performing the cast only in the call to `SOCKETS_SetSockOpt`?

Reply:
fixed in 0b96fcdce6391b1fe64a529091bf2c73cc0eb886